src/nvim/ex_cmds.c

```diff
@@ -3886,9 +3886,25 @@ static int do_sub(exarg_T *eap, const proftime_T timeout, const int cmdpreview_n
           goto skip;
         }
 
+        // If the substitute string contains a call to the 'input' function,
+        // prevent it from executing while the user is typing the command
+        // by skipping step 3.
+        bool skip_sub= false;
+        regmatch_T input_match = {
+           .regprog = vim_regcomp("\\\\=input(.*)", RE_MAGIC)
```

Comment:
This seems like a fragile hack that is bound to eventually cause a bug report. What about this suggestion: https://github.com/neovim/neovim/issues/11940#issuecomment-590139118

> Perhaps during the preview it should show dummy input like `INPUT`.

Maybe `input() could detect cmdpreview and return a hardcoded dummy string like "INPUT". Then when user hits enter, `input()` should actually get user input.